File: app/src/main/java/com/example/iwyatt/circleanimationdrawable/MainActivity.java
package com.example.iwyatt.circleanimationdrawable;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.ImageView;

import com.example.iwyatt.circleanimationdrawable.widgets.CircleAnimationDrawable;

public class MainActivity extends AppCompatActivity {

    private CircleAnimationDrawable mAnimationDrawable;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ImageView img = (ImageView) findViewById(R.id.img);

        mAnimationDrawable = new CircleAnimationDrawable(200);
        img.setImageDrawable(mAnimationDrawable);
    }

    @Override
    protected void onResume() {
        super.onResume();
        if (null != mAnimationDrawable) {
            mAnimationDrawable.start();
        }
    }

    @Override
    protected void onPause() {
        super.onPause();
        if (null != mAnimationDrawable) {
            mAnimationDrawable.stop();
        }
    }
}

File: kotlin/src/main/java/com/example/iwyatt/circleanimationdrawable/MainActivity.kt
package com.example.iwyatt.circleanimationdrawable

import android.app.Activity
import android.os.Bundle
import android.widget.ImageView
import com.example.iwyatt.circleanimationdrawable.widgets.CircleAnimationDrawable

/**
 * Created by wt on 17/5/25.
 */
class MainActivity : Activity() {

    var mAnimDrawable: CircleAnimationDrawable? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.layout_main)

        val img = findViewById(R.id.img) as ImageView

        mAnimDrawable = CircleAnimationDrawable(200.toFloat())

        img.setImageDrawable(mAnimDrawable)

    }

    override fun onResume() {
        super.onResume()
        mAnimDrawable?.start()
    }

    override fun onPause() {
        super.onPause()
        mAnimDrawable?.stop()
    }

}